pkg/ops/rawDiskGeneration.go

```diff
@@ -415,6 +418,48 @@ func (r *RawImage) createEFIPartitionImage() (string, error) {
 	return efiPartitionImage.File, nil
 }
 
+// createBiosPartitionImage creates a BIOS partition image with grub artifacts and installs grub to it
+func (r *RawImage) createBiosPartitionImage() (string, error) {
+	// Would need to format it and then copy the grub files to it
+	// Also install grub against it
+	// Remember to use the grub artifacts from the rootfs to support secureboot
+	// Create a temp dir for copying the files to
+	tmpDirBios := filepath.Join(r.TempDir(), "bios")
+	err := fsutils.MkdirAll(r.config.Fs, tmpDirBios, 0755)
+	if err != nil {
+		internal.Log.Logger.Error().Err(err).Str("target", tmpDirBios).Msg("failed to create temp dir")
+		return "", err
+	}
+	defer r.config.Fs.RemoveAll(tmpDirBios)
```

Comment:
We use the whole `TempDir` to create the bios.img file from it, but we only cleanup the `bios` and `recovery-mout` directories from it (with `defer`). The final artifact is still in the `TempDir`. It feels somewhat weird because it's not clear if the `TempDir` has anything else on it. Is the `TempDir` what becomes the rootfs in the end? If yes, is it ok that we also use it to store build artifacts like the `bios.img` file below?

Reply:
no, this is all wrong. Sorry about the confusing this seems to be a leftover from a thousand reworks. Bio.img is an empty diisk image, we dont need any of this mounts or anything, it just needs to be an empty image only with nothing inside as grub-install will then write core.img directly to that partition/